core/src/components/checkbox/test/bottom-content/checkbox.e2e.ts

```diff
@@ -0,0 +1,205 @@
+      const helperTextId = await helperText.getAttribute('id');
+      const ariaDescribedBy = await input.getAttribute('aria-describedby');
+
+      expect(ariaDescribedBy).toBe(helperTextId);
+    });
+    test('error text should be visible when checkbox is invalid', async ({ page }) => {
+      await page.setContent(
+        `<ion-checkbox class="ion-invalid ion-touched" helper-text="Helper text" error-text="Error text">Label</ion-checkbox>`,
+        config
+      );
+
+      const helperText = page.locator('ion-checkbox .helper-text');
+      const errorText = page.locator('ion-checkbox .error-text');
+      await expect(helperText).toBeHidden();
+      await expect(errorText).toBeVisible();
+      await expect(errorText).toHaveText('Error text');
+    });
+
+    test('input should have an aria-describedby attribute when error text is present', async ({ page }) => {
+      await page.setContent(
+        `<ion-checkbox class="ion-invalid ion-touched" helper-text="Helper text" error-text="Error text">Label</ion-checkbox>`,
+        config
+      );
+
+      const input = page.locator('ion-checkbox input[type=checkbox]');
+      const errorText = page.locator('ion-checkbox .error-text');
+      const errorTextId = await errorText.getAttribute('id');
+      const ariaDescribedBy = await input.getAttribute('aria-describedby');
+
+      expect(ariaDescribedBy).toBe(errorTextId);
+    });
+    test('input should have aria-invalid attribute when checkbox is invalid', async ({ page }) => {
+      await page.setContent(
+        `<ion-checkbox class="ion-invalid ion-touched" helper-text="Helper text" error-text="Error text">Label</ion-checkbox>`,
+        config
+      );
+
+      const input = page.locator('ion-checkbox input[type=checkbox]');
+
+      await expect(input).toHaveAttribute('aria-invalid');
+    });
+    test('input should not have aria-invalid attribute when checkbox is valid', async ({ page }) => {
+      await page.setContent(
+        `<ion-checkbox helper-text="Helper text" error-text="Error text">Label</ion-checkbox>`,
+        config
+      );
+
+      const input = page.locator('ion-checkbox input[type=checkbox]');
+
+      await expect(input).not.toHaveAttribute('aria-invalid');
+    });
+    test('input should not have aria-describedby attribute when no hint or error text is present', async ({ page }) => {
+      await page.setContent(`<ion-checkbox>Label</ion-checkbox>`, config);
+
+      const input = page.locator('ion-checkbox input[type=checkbox]');
+
+      await expect(input).not.toHaveAttribute('aria-describedby');
+    });
+  });
+});
+
+/**
+ * Rendering is different across modes
+ */
+configs({ modes: ['ios', 'md'], directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('checkbox: helper text rendering'), () => {
+    // Check the default label placement, end, and stacked
+    [undefined, 'end', 'stacked'].forEach((labelPlacement) => {
```

Comment:
What was the reasoning to only checking for these placements?

Reply:
The layout of the helper text wasn't different for `'fixed'` compared to `undefined` so I left it off, but I can add it if you think we should check it:

| `undefined` | `'fixed'` |
| --- | --- |
| ![checkbox-default](https://github.com/user-attachments/assets/b4d91677-0bb8-4b92-8064-9166a97b31be) | ![checkbox-fixed](https://github.com/user-attachments/assets/901e736e-6760-427c-91d8-183ce1b78750) |

Reply:
I'm fine with this. Thanks!